src/libcore/iter/iterator.rs

```diff
@@ -25,6 +25,10 @@ fn _assert_is_object_safe(_: &Iterator<Item=()>) {}
 /// generally, please see the [module-level documentation]. In particular, you
 /// may want to know how to [implement `Iterator`][impl].
 ///
+/// Note: Methods on infinite iterators that generally require traversing every
+/// element to produce a result may not terminate, even on traits for which a
+/// result is determinable in finite time.
```

Comment:
Given that the trait's doc-comment is essentially just a redirect to the module docs, I don't think this note needs to be here as well.  Maybe mention diverging in `fold` specifically, as the most general case of the methods that diverge when given infinite input?

Reply:
Yeah, I was divided 50:50 on this one. I've taken it out and replaced it with a comment on `fold` as you suggested.